src/main.rs

```diff
@@ -1 +1,214 @@
+        eprintln!("Error parsing config: {}", e);
+        process::exit(1);
+    });
+    setup_logger(config.log_level).unwrap_or_else(|e| {
+        eprintln!("Error setting up logger: {}", e);
+        process::exit(1);
+    });
+
+    let mut data_dir = config.data_dir.unwrap_or_else(|| {
+        config_folder_path().unwrap_or_else(|e| {
+            eprintln!("Error getting default data directory: '{}'.", e);
+            process::exit(1);
+        })
+    });
+    data_dir.push(config.bitcoind_config.network.to_string());
+    log::info!("Using data directory at '{}'.", data_dir.to_string_lossy());
+    if !data_dir.as_path().exists() {
+        log::info!("Data directory doesn't exist, creating it.");
+        create_datadir(&data_dir).unwrap_or_else(|e| {
+            eprintln!("Error creating data directory: '{}'.", e);
+            process::exit(1);
+        });
+    }
+    data_dir = fs::canonicalize(data_dir).unwrap_or_else(|e| {
+        eprintln!("Error canonicalizing data directory: '{}'.", e);
+        process::exit(1);
+    });
+
+    log::info!("Setting up the database");
+    let mut db_path = data_dir.clone();
+    db_path.push(path::Path::new(DATABASE_FILENAME));
+    setup_db(
+        &db_path,
+        &config.scripts_config.deposit_descriptor,
+        &config.scripts_config.unvault_descriptor,
+        &config.scripts_config.cpfp_descriptor,
+        config.bitcoind_config.network,
+    )
+    .unwrap_or_else(|e| {
+        log::error!("Error setting up database: '{}'", e);
+        process::exit(1);
+    });
+
+    log::info!("Setting up bitcoind connection");
+    let mut vault_watchonly_path = data_dir
+        .to_str()
+        .expect("Data dir must be valid unicode")
+        .to_string();
+    vault_watchonly_path.push_str(VAULT_WATCHONLY_FILENAME);
+    let bitcoind = start_bitcoind(&config.bitcoind_config, vault_watchonly_path.clone())
+        .unwrap_or_else(|e| {
+            log::error!("Error setting up bitcoind RPC connection: '{}'", e);
+            process::exit(1);
+        });
+
+    log::info!("Checking if bitcoind is synced");
+    wait_bitcoind_synced(&bitcoind);
+
+    load_watchonly_wallet(&bitcoind, vault_watchonly_path).unwrap_or_else(|e| {
+        log::error!("Error loading vault watchonly wallet: '{}'", e);
+        process::exit(1);
+    });
+    // TODO: load feebumping wallet too.
+
+    let mut noise_secret_path = data_dir.clone();
+    noise_secret_path.push(path::Path::new(NOISE_KEY_FILENAME));
+    log::info!(
+        "Reading or generating Noise key at '{:?}'",
+        noise_secret_path
+    );
+    let noise_secret = read_or_create_noise_key(&noise_secret_path).unwrap_or_else(|e| {
+        log::error!("Error reading or generating Noise key: '{}'", e);
+        process::exit(1);
+    });
+    log::info!(
+        "Using Noise key '{}'.",
+        NoisePubKey(curve25519::scalarmult_base(&curve25519::Scalar(noise_secret.0)).0)
```

Comment:
nit: why not return `Ok(NoisePubKey, NoisePrivKey)` with `read_or_create_noise_key` and avoid using scalarmult_base?

Reply:
We'd need `scalarmult_base` anyways but what we can do is simply use the `public_key()` helper of `SecretKey`, added a commit doing that to #3.